doc/sphinx/cligroup/cligroup.py

```diff
@@ -0,0 +1,51 @@
+from docutils import nodes
+from sphinx import addnodes
+from sphinx.directives import ObjectDescription
+from sphinx.util.compat import Directive
+from sphinx.util.docfields import Field
+
+cli_field_types = [
+        Field('summary', label='Summary', has_arg=False,
+                   names=('summary', 'shortdesc')),
+        Field('description', label='Description', has_arg=False,
+                   names=('description', 'desc', 'longdesc'))
+    ]
+
+class CliBaseDirective(ObjectDescription):
+    def handle_signature(self, sig, signode):
+        signode += addnodes.desc_addname(sig, sig)
+        return sig
+
+    def needs_arglist(self):
+        return False
+
+    def add_target_and_index(self, name, sig, signode):
+        signode['ids'].append(name)
+
+    def get_index_text(self, modname, name):
+        return name
+
+class CliGroupDirective(CliBaseDirective):
+    doc_field_types = cli_field_types
+
+class CliCommandDirective(CliBaseDirective):
+    doc_field_types = cli_field_types
+
+class CliArgumentDirective(CliBaseDirective):
+    doc_field_types = [
+        Field('summary', label='Summary', has_arg=False,
```

Comment:
let's factor our the summary and description since we have it above.  Something similar to:
def __init__(self):
        CliArgumentDirective.doc_field_types = cli_field_types.copy()
        CliArgumentDirective.doc_field_types.append(...values text here...)